### ISSUE TITLE
ci builds and uses vendor on ubuntu non-vendor runs

### ISSUE THREAD

**Issue by martinjrobins** [Chris00/sundials-sys#5]
For all the CI runners on ubuntu and for the non-vendor runs (e.g. `stable`), the `build.rs` builds and uses the vendor sundials rather than the system sundials. 

This is due to the ubuntu system sundials being version 5, but 6 is enforced. The `build.rs` warning is:

```
warning: sundials-sys@0.4.0: System sundials version = 5 < 6, will use the vendor version
```

**Comment by Chris00** [comment]
Thanks for pointing that out.  At least that shows that the fallback works! 😇  Do you see anything to do but waiting for a newer Ubuntu version to become LTS?  (Although not ideal, on my Debian machine, I use the system libraries so that part gets tested.)

**Comment by martinjrobins** [comment]
depends. What are you plans in terms of what sundials versions you want to support? At the moment you are testing against just two: whatever is in `ubuntu-latest` and the version that your submodule is pointing to, which is fine but you are a bit at the mercy of however ubuntu has configured its sundials package. You could setup your CI to download and install (in a standard system location) a specific version or range of versions to test against. Its a bit more work but then you can state in the docs: "sundials-sys is testing against sundials version X to Y" to give users a good idea of what versions are guarenteed to work.

**Comment by Chris00** [comment]
That's a good idea.  I've added testing against different versions (with some fixes for v7.0.0).  These fixes use code from [your KLU PR](https://github.com/Chris00/sundials-sys/pull/6) so it is now incompatible.
